### ISSUE TITLE
Script.log not found

### ISSUE THREAD

**Issue by raphael-shaloa** [Thelukepet/Khazad-Voice-TTS#8]
I end up with Script.log not found even though it does exist and logs the NPC names. Here is my box:

    ========================================
       LOTRO NARRATOR - AI VOICE OVER
    ========================================

[SELECT AUDIO ENGINE]
1. CPU (Kokoro) [Default]
   -> Fast, Reliable. Works on all PCs.
2. GPU (LuxTTS)
   -> Higher Quality. REQUIRES NVIDIA GPU.

Enter choice (1 or 2): 2
18:35:08 - [src.db] - INFO - ✅ Database loaded: 6373 NPCs found.
18:35:08 - [TTS_FACTORY] - INFO - 🚀 User selected GPU (LuxTTS). Initializing...
18:35:08 - [src.tts.lux] - INFO - Loading LuxTTS Model on cuda...
WARNING:root:Failed import k2 with error No module named 'k2'. Swoosh functions will fallback to PyTorch implementation, leading to slower speed and higher memory consumption.
H:\Programmes\Khazad-Voice-TTS-1.2\venv\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
Fetching 11 files: 100%|███████████████████████████████████████████████████████████████████████| 11/11 [00:00<?, ?it/s]
Device set to use cuda
H:\Programmes\Khazad-Voice-TTS-1.2\LuxTTS\zipvoice\modeling_utils.py:150: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  vocos.load_state_dict(torch.load(f"{model_path}/vocoder/vocos.bin", map_location=params.device))
Loading model on GPU
18:35:17 - [src.tts.lux] - INFO - ✅ LuxTTS Ready. Loaded 83 reference voices.
INFO:src.tts.lux:✅ LuxTTS Ready. Loaded 83 reference voices.
18:35:17 - [src.tts.lux] - INFO - 🔥 Warming up LuxTTS (takes ~10s for first run)...
INFO:src.tts.lux:🔥 Warming up LuxTTS (takes ~10s for first run)...
H:\Programmes\Khazad-Voice-TTS-1.2\venv\Lib\site-packages\linacodec\vocoder\vocos.py:132: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=False):
18:35:24 - [src.tts.lux] - INFO - 🔥 Warmup Complete in 7.07s.
INFO:src.tts.lux:🔥 Warmup Complete in 7.07s.
18:35:24 - [ENGINE] - INFO - 🧠 Loaded Memory for mode: retail
INFO:ENGINE:🧠 Loaded Memory for mode: retail

[RETAIL MODE STARTED]
Watching Log: C:\Users\Raphael\Documents\The Lord of the Rings Online\Script.log
1. Ensure 'getNPCNames' LOTRO plugin is installed and loaded for your character.
18:35:24 - [WATCHER] - WARNING - Log file C:\Users\Raphael\Documents\The Lord of the Rings Online\Script.log not found - waiting...
WARNING:WATCHER:Log file C:\Users\Raphael\Documents\The Lord of the Rings Online\Script.log not found - waiting...

**Comment by Thelukepet** [comment]
Hi, thanks for reaching out! Strange, it looks like it can indeed find the file. Will look into it.  Could you try deleting the .log file and unloading then loading the lotro plugin again?

**Comment by raphael-shaloa** [comment]
I also notice those two warnings

WARNING:root:Failed import k2 with error No module named 'k2'. Swoosh functions will fallback to PyTorch implementation, leading to slower speed and higher memory consumption.
H:\Programmes\Khazad-Voice-TTS-1.2\venv\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)

Is it normal or is it something else that needs fixing on my end?

**Comment by Thelukepet** [comment]
The k2 warning is something I will look in to when I have the time. The other error shouldn’t be an issue but might be because the location where you run it from is not the C:/ drive.

I only tested with my system that has the LOTRO documents and the .log file from the plugin on my C:/ drive.

However for me it did work when e.g the program was running from my D:/ drive. 

Could you send the path of:
* the .log file which is filled through the plugin
* your lotro game files installation
* the location of where you are running the Khazad-Voice-TTS program from.
* finally, make sure the start.bat files are opened from inside the Khazad-Voice-TTS root directory (e.g. by double-clicking). It might not work when you call the .bat file through a different location.

**Comment by Thelukepet** [comment]
That should be the issue! If you go to src/config.py you should change the SCRIPT_LOG to the following:

SCRIPT_LOG = r"H:\Raphael\Mes documents\The Lord of the Rings Online\Script.log"

You probably also need to change the tesseract path to fit your situation. Because they are set to the default C:\ drive as well. Try to find your tesseract.exe and add the path to possible_paths.

possible_paths = [
    r"C:\Program Files\Tesseract-OCR\tesseract.exe",
    r"C:\Program Files (x86)\Tesseract-OCR\tesseract.exe",
  r"C:\Users\admin\AppData\Local\Programs\Tesseract-OCR\tesseract.exe",
]

TESSERACT_CMD = r"C:\Program Files\Tesseract-OCR\tesseract.exe"  # Default fallback


I think that should fix your issue. In the near future I will ask users beforehand what their paths to these things are so we can prevent this  in the future. Thanks a lot for spotting this!

**Comment by raphael-shaloa** [comment]
It did fix that issue, thanks!

I switched to CPU because there was too big of a delay with GPU. I had to delete the npc_memory .json file like you said to the other guy. There is still a small delay with CPU but it's faster.

The other "problem" I had is calibrating does not work with my JRR skin. I think it's because it didn't have leaves. I have to put the original UI for it to work. I think a static place like earlier versions would have been preferable for that case.

I completely understand if you don't want to bother making your tool work with skins though.

**Comment by Thelukepet** [comment]
Hey, you can try to reduce the `TTS_WAVE_STEPS` in the `src/config.py` from 10 to 4 this should significantly boost the speed of the calculations without losing too much quality. For me it speeds up the GPU processing from 2-3s to about 1 second
